DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateGroupsController.cs

```diff
@@ -32,12 +32,35 @@ public IActionResult Index()
         [Route("{groupId:int}/Delegates/{page:int=1}")]
         public IActionResult GroupDelegates(int groupId, int page = 1)
         {
+            var centreId = User.GetCentreId();
+            if (!groupsDataService.IsGroupAtUserCentre(groupId, centreId))
+            {
+                return NotFound();
+            }
+            
             var groupName = groupsDataService.GetGroupNameForGroupId(groupId);
             var groupDelegates = groupsDataService.GetGroupDelegates(groupId);
 
             var model = new GroupDelegatesViewModel(groupId, groupName, groupDelegates, page);
 
             return View(model);
         }
+
+        [Route("{groupId:int}/Courses/{page:int=1}")]
+        public IActionResult GroupCourses(int groupId, int page = 1)
+        {
+            var centreId = User.GetCentreId();
+            if (!groupsDataService.IsGroupAtUserCentre(groupId, centreId))
+            {
+                return NotFound();
+            }

```

Comment:
Would be good to add controller tests for the NotFound functionality (here + delegates)